src/Controller/NonPaginatedEntityListingControllerTrait.php

```diff
@@ -50,7 +50,7 @@ public function getEntities(): array
     /**
      * @inheritdoc
      */
-    abstract protected function responseToArray(ResponseInterface $response): array;
+    abstract protected function responseToArray(ResponseInterface $response, bool $expandCompatibility = false): array;
```

Comment:
Wow, this is a major change... why a response to array converter have to know about whether the org supports expand or not? I have to think about how this could be solved properly...

Reply:
Maybe we should rather introduce a `responseToArrayHyrid()` method and call that instead of adding too much complexity to the existing method. Where `responseToArray()` is called you already have more context (see above) so you can decide which method should be called.

Reply:
Btw, the returned structure is kinda similar now to the structure that Apigee Edge returns when `expand=true`, the major difference is the nested items are not objects just ids. See `\Apigee\Edge\Controller\PaginationHelperTrait::listEntitiesWithoutCps()` for example, where `responseToArray()` first parse the complete JSON to an array, then we drop the entity type-specific key ("proxies" in this example) and then iterate over the second level array items

```php
[
    {
      "name": "helloworld"
    },
    {
      "name": "weather"
    }
  ]
```

to parse them to entity objects. In this case. we would not parse them entity objects, we would only return the nested values with an `array_reduce()` call.

Reply:
I added a helper method `normalizeExpandFalseForHybrid()` to separate this logic further.

Reply:
I am good w/the approach of having a expandCompatibility param.

Reply:
> Maybe we should rather introduce a `responseToArrayHyrid()` method and call that instead of adding too much complexity to the existing method. Where `responseToArray()` is called you already have more context (see above) so you can decide which method should be called.

Btw, I considered it, but the "appropriate context" to figure out whether to use `responseToArray()` or `responseToArrayHybrid()` might end up being some spaguetti code, as it might be several calls above - it's not just where `responseToArray()` is called, but where we the query param expand=false is used, so I feel the current approach is more practical.